charts/komodor-agent/templates/configmap.yaml

```
@@ -39,5 +39,21 @@ data:
     resources:
       {{- toYaml .Values.allowedResources | trim | nindent 6 }}
 
+    {{- if .Values.tags }}
+    {{- if and .Values.tags (kindIs "map" .Values.tags) }}
+    tags: {{ .Values.tags | nindent 6 }}
+    {{- else if and .Values.tags (kindIs "string" .Values.tags) }}
+    {{ $tags := .Values.tags | splitList ";" }}
+    {{ $tagMap := dict }}
+    {{ range $tags }}
+      {{ $pair := splitList ":" . }}
+      {{ $key := index $pair 0 }}
+      {{ $value := index $pair 1 }}
+      {{ $_ := set $tagMap $key $value }}
```

Maybe add a check to ensure the string contains both a key and a value 
```suggestion
      {{- if eq (len $pair) 2 }}
        {{ $key := index $pair 0 }}
        {{ $value := index $pair 1 }}
        {{ $_ := set $tagMap $key $value }}
      {{- end }}
```